src/runtime/plugin.ts

```diff
@@ -21,18 +22,23 @@ export default defineNuxtPlugin((nuxtApp) => {
         ...headers,
       }
     },
-    fetch: (input, options) =>
-      globalThis.$fetch.raw(input.toString(), options)
-        .catch((e) => {
-          if (e instanceof FetchError && e.response)
-            return e.response
-
-          throw e
-        })
-        .then(response => ({
+    fetch: async (input, options) => {
+      try {
+        const response = await globalThis.$fetch.raw(input.toString(), options)
+
+        return {
           ...response,
           json: () => Promise.resolve(response._data),
-        })),
+        }
+      }
+      catch (e) {
+        if (e instanceof FetchError && e.response)
+          return e.response
+
+        throw e
+      }
+    },
+    transformer: config.transformer,
```

Comment:
Nuxt will not be able to serialize this.

```bash
[nitro] [dev] [uncaughtException] TypeError: rt.transformer.serialize is not a function
```

Have you tried running this locally?

Reply:
Yes, I did and it worked flawlessly. I've no idea why it suddenly broke. Nuxt and/or vite just doesn't seem to pass through functions correctly anymore. I've no idea what is going on. I think I'll close this PR and think about a way how to do this less hacky